src/cache_hash.cpp

```diff
@@ -24,6 +24,13 @@ void cache_init(CacheHash *ch, Buf *manifest_dir) {
     ch->b64_digest = BUF_INIT;
 }
 
+void cache_mem(CacheHash *ch, const char *ptr, size_t len) {
+    assert(ch->manifest_file_path == nullptr);
+    assert(ptr != nullptr);
+    // + 1 to include the null byte
+    blake2b_update(&ch->blake, ptr, len + 1);
```

Comment:
In this case because the values happen to come from a zig null terminated pointer, there is a null byte. But this C++ API probably should not assume this. Instead it should issue 2 `blake2b_update` calls, one for the memory, and then another to write a null byte. However, given that the payload in this case can have null bytes, I'm not sure how valuable that is anyway. I think it should just be `len` instead of `len + 1`.

Also: unnecessary, but just as a cleanup, cache_str can be simplified to call `cache_mem(ch, ptr, strlen(ptr) + 1)`

Reply:
incorporated